core/src/main/java/org/apache/iceberg/ManifestFiles.java

```diff
@@ -300,4 +328,14 @@ private static ManifestFile copyManifestInternal(
 
     return writer.toManifestFile();
   }
+
+  private static InputFile newInputFile(FileIO io, String path, long length) {
+    InputFile inputFile = io.newInputFile(path, length);
```

Comment:
I have seen cases where calling `newInputFile` was the bottleneck, since the HadoopFileIo created a new `FileSystem` object every time. Why did we decide to key the cache with the `inputFile` instead of the `path`?

Reply:
`inputFile` is not used as cache key, but rather as a fallback within `CachingInputFile` class (field `wrappedInputFile `) if `path` does not exist in `ContentCache` yet.

I understand that `HadoopInputFile.fromLocation()` will create a new `FileSystem` upon initialization. Maybe I can tweak the code to lazily instantiate `wrappedInputFile` by passing `io`, `path`, and `length` instead as constructor parameters of `CachingInputFile`?

Reply:
[d86461e](https://github.com/apache/iceberg/pull/4518/commits/d86461ece6643f86a051b4e38fa00d76bcec31d1) address this improvement.